modules/client/src/test/java/org/apache/ignite/client/fakes/FakeCursor.java

```diff
@@ -114,6 +114,11 @@ public CompletableFuture<Void> closeAsync() {
         return nullCompletedFuture();
     }
 
+    @Override
+    public CompletableFuture<Void> cancelAsync(CancellationReason reason) {
+        return closeAsync();
```

Comment:
Do we have a plan for supporting cancellation on client?

Reply:
If you are talking about CancelHandle, then yes, we will support it eventually.

But this particular method is part of internal API of sql engine. It's supposed to be a server side API, providing an ability to terminate query with certain reason. Depending on the reason, we chose the way how to release acquired resources. For instant, in case of transaction, we do commit  transaction if reason is CLOSE, and rollback transaction otherwise.